src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -420,6 +420,36 @@ private void initialize() {
         registerActiveQueueRegistrationRefresh();
         registerQueueCheck();
         registerMetricsGathering(configuration);
+        registerNotExpiredQueueCheck();
+    }
+
+    private void registerNotExpiredQueueCheck() {
+        vertx.setPeriodic(20 * 1000, event -> {
+            if (!log.isDebugEnabled()) {
+                return;
+            }
+            String keysPattern = consumersPrefix + "*";
+            log.debug("RedisQues list not expired consumers keys:");
+            redisProvider.redis().onSuccess(redisAPI -> redisAPI.keys(keysPattern, keysResult -> {
```

Comment:
I think we should not use the `keys` command in production redis databases since this is a blocking command. The more consumer-keys are stored, the longer this command takes. This check should probably implemented with the `scan` command.

Reply:
OK , thx, done